debezium-embedded/src/main/java/io/debezium/embedded/EmbeddedEngine.java

```diff
@@ -808,10 +838,32 @@ public Map<String, String> configs() {
                                 break;
                             }
                             catch (RetriableException e) {
-                                LOGGER.info("Retrieable exception thrown, connector will be restarted", e);
-                                // Retriable exception should be ignored by the engine
-                                // and no change records delivered.
-                                // The retry is handled in io.debezium.connector.common.BaseSourceTask.poll()
+                                int maxRetries = getErrorsMaxRetries();
+                                LOGGER.info("Retriable exception thrown, connector will be restarted; errors.max.retries={}", maxRetries, e);
+                                if (maxRetries != 0) {
```

Comment:
@jpechane In case the maxRetries is 0, it does not go thorough the retry logic, however it does not break the while loop either and keeps falling back as RetriableException. I believe we need a else part to this condition:

```
else {
    handlerError = e;
    break;
}
```

Reply:
Even when maxRetries is non-zero, say `errors.max.retries=1`, it keeps looping over and resets totalRetries to 0 each time, resulting in retrying forever.

Reply:
@poonam-meghnani Hi. This mechanism is complementary to the one inside the poll() method of the BaseSourceTask. In the poll() method, the first line calls startIfNeededAndPossible() which calls start() method which for some connectors can throw non retriable exceptions or wrapped retriable exceptions. In that case the engine was stopping. Now with the new mechanism, the connector will try to restart no matter what the exception in the start() method is. 
Now back to your questions, when max retries is 0, the new mechanism is disabled and the behavior is exactly as it was before. You could check the comments above to understand the different behavior of different connectors without this new mechanism.

Reply:
The behaviour that we're observing is that even with max retries 0, the connector still retries in an infinite loop. Is this an expected behaviour? It could be that it was broken in this way previously?

Reply:
@alwaysbemark like I said, 0 max retries disables only this new mechanism, but not the one implemented by this issue: https://issues.redhat.com/browse/DBZ-5244. Currently both of the mechanisms are in place. Depending on the connector, DBZ-5244 is able to retry by itself to restart the task or not. Where this is not enough, the new mechanism does the job.